src/gausskernel/storage/access/k2/pg_gate_api.cpp

```diff
@@ -200,14 +200,19 @@ K2PgStatus PgGate_ExecCreateDatabase(const char *database_name,
                                  K2PgOid next_oid) {
   elog(LOG, "PgGateAPI: PgGate_ExecCreateDatabase %s, %d, %d, %d",
          database_name, database_oid, source_database_oid, next_oid);
-  return K2PgStatus::NotSupported;
+  return pg_gate->GetCatalogClient()->CreateDatabase(database_name,
+      k2pg::PgObjectId::GetDatabaseUuid(database_oid),
+      database_oid,
+      source_database_oid != k2pg::kPgInvalidOid ? k2pg::PgObjectId::GetDatabaseUuid(source_database_oid) : "",
+      "" /* creator_role_name */, next_oid);
 }
 
 // Drop database.
 K2PgStatus PgGate_ExecDropDatabase(const char *database_name,
                                    K2PgOid database_oid) {
-  elog(DEBUG5, "PgGateAPI: PgGate_ExecDropDatabase %s, %d", database_name, database_oid);
-  return K2PgStatus::NotSupported;
+    elog(DEBUG5, "PgGateAPI: PgGate_ExecDropDatabase %s, %d", database_name, database_oid);
+    return pg_gate->GetCatalogClient()->DeleteDatabase(database_name,
+        k2pg::PgObjectId::GetDatabaseUuid(database_oid));
```

Comment:
same here, check uuid or database_oid string.

Reply:
Same as above. Both database oid and uuid are passed. The collection is not deleted but the meta table deleted https://github.com/futurewei-cloud/chogori-sql/blob/master/src/k2/connector/pggate/catalog/sql_catalog_manager.cc#L342